crypto/stark/src/autogen_sol.rs

```diff
@@ -0,0 +1,199 @@
+// TODO - Format the hash map returns to remove this warning
+#![allow(unused_results)]
+use crate::rational_expression::*;
+use std::{
+    cmp::Ordering, collections::BTreeMap, fs::File, io::prelude::*, path::Path, prelude::v1::*,
+};
+
+pub fn autogen2(
+    _trace_len: usize,
+    _public: &[&RationalExpression],
+    constraints: &[RationalExpression],
+    _n_rows: usize,
+    n_cols: usize,
+    blowup: usize,
+) {
+    let mut traces = BTreeMap::new();
+
+    for exp in constraints.iter() {
+        traces.extend(exp.trace_search());
```

Comment:
Move the `trace_arguments`, `soldity_encode` and the `*_search` functions to this file. You can add a `impl RationalExpression {` block here (Rust allows many `impl` blocks that don't have to be in the same module)

Reply:
Unless we plan to deprecate those in favor of DAG later, in which case we can just remove them